### ISSUE TITLE
In com.netflix.iceberg.pig.SchemaUtil does not convert complex schemas in maps and lists to Pig Schema

### ISSUE THREAD

**Issue by rdsr** [Netflix/iceberg#56]
I was trying to reproduce some of the potential issues I highlighted in PigStorage pull request when I hit a possible bug in SchemaUtil

1.  In com.netflix.iceberg.pig.SchemaUtil#convertComplex  If the input is a list or a map, for their nested elements we call the method convert(Type type) which does not seem to take into account if its input is a complex schema. It seems that convert(NestedField)  is much closer to what we want, isn't it? I tried to make the changes, but I'm not sure now to get a NestedField out of a list or a map value type. The methods are there but they seem to expect an "id" [com.netflix.iceberg.types.Types.MapType#field(int id)]

Here's a test case which shows that the Pig schema does not contain the nested struct 
```
@Test
  public void testTupleInMap() throws IOException {
    Schema icebergSchema = new Schema(
        optional(
            1, "nested_list",
            MapType.ofOptional(
                2, 3,
                StringType.get(),
                ListType.ofOptional(
                    4, StructType.of(
                        required(5, "id", LongType.get()),
                        optional(6, "data", Types.StringType.get()))))));

    ResourceSchema pigSchema = SchemaUtil.convert(icebergSchema);
    Assert.assertEquals("nested_list:[[]]", pigSchema.toString()); // The output should contain a nested struct within a list within a map, I think.
  }
```

**Comment by danielcweeks** [comment]
@rdsr Thanks for pointing this out.  You are correct, the nested schemas are not being converted.  

Interestingly, pig does produce the data correctly (i.e. the converters appear to be working and pig doesn't appear to mind that the data produced does not match the converted schema).

I've got a fix for this and will post that later (hopefully with some tests).

**Comment by danielcweeks** [comment]
@rdsr I'm updated and added your test.  I'm not perfectly clear on how pig represents nested schemas, but I believe this new approach is accurate. See  #57

**Comment by rdsr** [comment]
thanks @danielcweeks . I'll have a look.

**Comment by rdblue** [comment]
I think this was fixed in #58.
